File: Data Structures/Stacks/C/StackUsingLL.c
#include<stdio.h>
#include<stdlib.h>

struct node{
    int data;
    struct node *next;
};

struct node *top=NULL;

struct node *createNode(int value)
{
    struct node *newnode= (struct node*)malloc(sizeof(struct node));
    newnode->next=NULL;
    newnode->data =value;

    return newnode;
}

void push(value)
{
    struct node *newnode = createNode(value);
    if (top==NULL)
    {
        top=newnode;
    }
    else
    {
        newnode->next=top;
        top=newnode;
    }
}

void pop()
{
    if (top==NULL)
    {
        printf("Stack Underflow\n");
    }
    else{
        struct node *temp=top;
        top=top->next;
        free(temp);
    }
}
    

void menu(){
    printf("1. Push Element in stack \n");
    printf("2. Pop the element \n");
    printf("3. Print stack\n");
    printf("4. Exit()\n");
}

void show()
{
    struct node *temp=top;
    while (temp!=NULL)
    {
        printf("%d\t",temp->data);
        temp=temp->next;
    }
    printf("\n");
    
}

int main()
{
    int ch,value;
    while (1)
    {
        menu();
        printf("Enter your choice : ");
        scanf("%d",&ch);
        switch (ch)
        {
        case 1:
            /* Push */
            printf("Enter the value to push : ");
            scanf("%d",&value);
            push(value);
            break;
        case 2:
            /* Pop */
            pop();
            break;
        case 3:
            /* print */
            show();
            break;
        case 4:
            /* exit */
            exit(0);
            break;
        
        default:
            break;
        }

    }
    
}

File: Data Structures/Stacks/python/stackUsingLL.py
class node:
    def __init__(self,data):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.top = None
    
    def Push(self,data):
        newnode = node(data)
        if self.top == None:
            self.top = newnode
        else:
            newnode.next = self.top
            self.top = newnode

    def Pop(self):
        if self.top ==None:
            print('Stack Underflow')
        else:
            self.top = self.top.next
            print('Element poped')    
    
    def Show(self):
        temp= self.top
        while temp is not None:
            print(temp.data,'5')
            temp = temp.next

def menu():
    print('*********************************************************************')
    print()
    print("1. Push Element in stack ")
    print("2. Pop the element ")
    print("3. Print stack")
    print("4. Exit()")
    print()
    print('*********************************************************************')


if __name__ =='__main__':
    while True:
        menu()
        ll = Stack()
        ch = int(input('Enter your choice : '))
        
        if ch==1:
            value = int(input('Enter the value to push : '))
            ll.Push(value)    
        elif ch==2:
            ll.Pop()
        elif ch==3:
            ll.Show()
        elif ch==4:
            exit()
        else:
            print('invalid option selected')
